dagmc_stats/dagmc_stats.py

```diff
@@ -471,12 +471,71 @@ def get_roughness(my_core, native_ranges):
 
     outputs
     -------
-    roughness : (numpy array) the roughness for all surfaces in the meshset
+    roughness : (dictionary) the roughness for all vertices in the meshset
     """
     tri_vert_data, all_verts = get_tri_vert_data(my_core,
                                                     native_ranges[types.MBTRI])
     gc_all = get_gaussian_curvature(my_core, all_verts, tri_vert_data)
-    roughness = np.zeros(len(all_verts))
-    for vert_idx, vert_i in enumerate(all_verts):
-        roughness[vert_idx] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
+    roughness = {}
+    for vert_i in all_verts:
+        roughness[vert_i] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
+    if tag:
```

Comment:
Right now this will add the tag to the moab instance in memory but it doesn't look like it gets written to a file. I think that in `generate_stats.py` a new file should be written out if tags are added after all the stats calculations are finished. So there needs to be some way to determine if tags have been set. 

One thought is that the value `tag` can be returned from this function in `generate_stats.py`. If after all stats have been run, if `tag` is true, then write the file. Otherwise we don't need to write out a new file. This will allow other tags to be added to the geometry as well and just write out a file with all tags.